### PR TITLE
Restored AttributeUsage projection

### PR DESCRIPTION
**Pull request by Scottj1s**
Restored the AttributeUsage projection

### CODE REVIEW

File: src/cswinrt/code_writers.h [hunk 1]
```diff
@@ -1427,7 +1427,9 @@ remove => %.% -= value;
                 }
             }
             // Skip metadata attributes
-            if (attribute_namespace == "Windows.Foundation.Metadata" && attribute_name != "DefaultOverload" && attribute_name != "Overload") continue;
+            if (attribute_namespace == "Windows.Foundation.Metadata" &&
```
Comment (j0shuams) [inline src/cswinrt/code_writers.h:1430]: What do you think of moving the check to a predicate and calling that the function from the `if ()` ?

File: src/cswinrt/code_writers.h [hunk 2]
```diff
@@ -1427,7 +1427,9 @@ remove => %.% -= value;
                 }
             }
             // Skip metadata attributes
-            if (attribute_namespace == "Windows.Foundation.Metadata" && attribute_name != "DefaultOverload" && attribute_name != "Overload") continue;
+            if (attribute_namespace == "Windows.Foundation.Metadata" && 
+                attribute_name != "DefaultOverload" && attribute_name != "Overload" && attribute_name != "AttributeUsage") 
+                continue;
             attributes[attribute_full] = std::move(params);
         }
         if (auto&& usage = attributes.find("AttributeUsage"); usage != attributes.end())
```
Comment (manodasanW) [inline src/cswinrt/code_writers.h:1435]: Any concerns with the below not finding anything given that would probably get skipped:

            bool allow_multiple = attributes.find("Windows.Foundation.Metadata.AllowMultiple") != attributes.end();

Comment (Scottj1s) [reply src/cswinrt/code_writers.h:1435]: yup - fix coming for that

File: src/cswinrt/code_writers.h (hunk 1 above)
Comment (Scottj1s) [reply src/cswinrt/code_writers.h:1430]: yes, that kind of factoring could be helpful if called multiple times.  here, probably just adds unnecessary complexity.

**Review by manodasanW — APPROVED** [review]

**Pull request merged**